spinalcordtoolbox/scripts/sct_create_mask.py

```diff
@@ -242,23 +242,14 @@ def create_mask(param):
     for iz in range(0, nz, 1):
         if iz in z_centerline_not_null:
             cx[iz], cy[iz] = ndimage.measurements.center_of_mass(np.array(data_centerline[:, :, iz]))
-    # create 2d masks
-    file_mask = 'data_mask'
+
+    # create 2d masks at each axial slice
+    im_out = zeros_like(im_data, dtype=np.float64)
     for iz in range(nz):
-        if iz not in z_centerline_not_null:
-            # write an empty nifty volume
-            img = nibabel.Nifti1Image(data_centerline[:, :, iz], None, hdr)
-            nibabel.save(img, (file_mask + str(iz) + '.nii'))
-        else:
+        if iz in z_centerline_not_null:
             center = np.array([cx[iz], cy[iz]])
-            mask2d = create_mask2d(param, center, param.shape, param.size, im_data=im_data)
-            # Write NIFTI volumes
-            img = nibabel.Nifti1Image(mask2d, None, hdr)
```

Comment:
The previous CI error was due to 5d8ec15. But, now that that's been fixed, new tests are [failing](https://github.com/neuropoly/spinalcordtoolbox/pull/3270/checks?check_run_id=2103148815#step:3:3540). However, I believe this is because the old mask method was faulty (and the tests were based off of the old mask method).

I noticed that there were slight differences between `mask.nii` generated by the new method, and `mask.nii` generated by the old method:

```markdown
### Near the center
mask_old
[22 21 2]: 0.7999999998137355
mask_new
[22 21 2]: 0.8007324328646064

### Near the edge
mask_old
[32 21 2]: 0.0
mask_new
[32 21 2]: 0.0003356984816491604
```

This reminded me of #3232, so I checked the old method to see the datatypes:

```python
# Context: img = nibabel.Nifti1Image(mask2d, None, hdr)
>>> mask2d.dtype
dtype('float64')
>>> hdr.get_data_dtype()
dtype('uint8')
```

Aha. It's the same issue as #3232. The datatypes don't match, and the float values get mangled by nibabel's scaling method.

Now, I believe the new method fixes the problem via 5d8ec15. But, now the tests have [outdated ground truth](https://github.com/neuropoly/spinalcordtoolbox/pull/3270/checks?check_run_id=2103148815#step:3:3540), so they should be updated.

Reply:
(As an aside, this is another instance showing why we should be validating header properties throughout SCT. Lately, the header has been a big source of bugs that have silently slipped through, considering #2454, #3174, #3232, and #3254.)

Reply:
> But, this would mean the tests have outdated ground truth, and should be updated.

Fixed in 5e5d655.